packages/api/src/controllers/webhook.ts

```diff
@@ -241,28 +241,57 @@ app.post(
   async (req, res) => {
     const { id } = req.params;
     const webhook = await db.webhook.get(id);
-    if (!webhook) {
-      return res.status(404).json({ errors: ["not found"] });
+    if (!webhook || webhook.deleted) {
+      return res.status(404).json({ errors: ["webhook not found or deleted"] });
     }
 
-    const doc = req.body.status;
-    const webhookResponse = req.body.response;
+    const { response, errorMessage } = req.body as WebhookStatusPayload;
 
-    if (!doc) {
-      return res.status(422).json({ errors: ["missing status in payload"] });
+    if (!response || !response.response) {
+      return res.status(422).json({ errors: ["missing response in payload"] });
+    }
+    if (!response.response.body && response.response.body !== "") {
+      return res
+        .status(400)
+        .json({ errors: ["missing body in payload response"] });
     }
 
-    await db.webhook.update(req.params.id, {
-      ...webhook,
-      status: doc,
-    });
-
-    if (webhookResponse) {
-      webhookResponse.id = uuid();
-      let w: WithID<WebhookResponse> = webhookResponse;
-      await db.webhookResponse.create(w);
+    try {
+      const triggerTime = response.createdAt ?? Date.now();
+      let status: DBWebhook["status"] = { lastTriggeredAt: triggerTime };
+      if (
+        response.statusCode >= 300 ||
+        !response.statusCode ||
+        response.statusCode === 0
+      ) {
```

Comment:
Response field should be optional since not always there will be a response, and `statusCode` should be obligatory (checked by JSON schema) and so we don't need to check if it's present here:
```suggestion
      if (
        errorMessage || 
        response?.statusCode >= 300
      ) {
```